### DISCUSSION TITLE
Audio doesn't work in windows players, when added to video

### DISCUSSION THREAD

**Discussion by NeighbouringPanda** [mifi/lossless-cut#708] [Support]
I added an extern audio track (opus) to a .mp4 video without audio (h264) and exported the result as .mp4 file.
Both tracks video and audio are activated in track menu.

**1.**   When opened in "Windows Foto App" or "windows media player" there is no audio, **there only is audio with vlc player**.
    I tested the issued media player by playing other videos with same codec and container format and it worked fine (audio active).

**2.**   This Issue also doesn't show up when exporting the video to .mkv format. Then the audio track is visible to all players...

Can you explain me what I'm doing wrong with adding/exporting the audio to video?
Or why this could be happening?

**Report:**
No error

{
  "state": {
    "filePath": "C:\\Test1 vid.mp4",
    "fileFormat": "mp4",
    "externalStreamFiles": {
      "C:\\Test1 audio.opus": {
        "streams": [
          {
            "index": 0,
            "codec_name": "opus",
            "codec_long_name": "Opus (Opus Interactive Audio Codec)",
            "codec_type": "audio",
            "codec_time_base": "1/48000",
            "codec_tag_string": "[0][0][0][0]",
            "codec_tag": "0x0000",
            "sample_fmt": "fltp",
            "sample_rate": "48000",
            "channels": 2,
            "channel_layout": "stereo",
            "bits_per_sample": 0,
            "r_frame_rate": "0/0",
            "avg_frame_rate": "0/0",
            "time_base": "1/1000",
            "start_pts": -7,
            "start_time": "-0.007000",
            "disposition": {
              "default": 1,
              "dub": 0,
              "original": 0,
              "comment": 0,
              "lyrics": 0,
              "karaoke": 0,
              "forced": 0,
              "hearing_impaired": 0,
              "visual_impaired": 0,
              "clean_effects": 0,
              "attached_pic": 0,
              "timed_thumbnails": 0
            },
            "tags": {
              "language": "eng"
            }
          }
        ],
        "formatData": {
          "filename": "C:\\Test1 audio.opus",
          "nb_streams": 1,
          "nb_programs": 0,
          "format_name": "matroska,webm",
          "format_long_name": "Matroska / WebM",
          "start_time": "-0.007000",
          "duration": "361.561000",
          "size": "5198875",
          "bit_rate": "115031",
          "probe_score": 100,
          "tags": {
            "encoder": "google/video-file"
          }
        }
      }
    },
    "mainStreams": [
      {
        "index": 0,
        "codec_name": "h264",
        "codec_long_name": "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10",
        "profile": "Main",
        "codec_type": "video",
        "codec_time_base": "1/120",
        "codec_tag_string": "avc1",
        "codec_tag": "0x31637661",
        "width": 1280,
        "height": 720,
        "coded_width": 1280,
        "coded_height": 720,
        "closed_captions": 0,
        "has_b_frames": 1,
        "sample_aspect_ratio": "1:1",
        "display_aspect_ratio": "16:9",
        "pix_fmt": "yuv420p",
        "level": 32,
        "color_range": "tv",
        "color_space": "bt709",
        "color_transfer": "bt709",
        "color_primaries": "bt709",
        "chroma_location": "left",
        "field_order": "progressive",
        "refs": 1,
        "is_avc": "true",
        "nal_length_size": "4",
        "r_frame_rate": "60/1",
        "avg_frame_rate": "60/1",
        "time_base": "1/15360",
        "start_pts": 0,
        "start_time": "0.000000",
        "duration_ts": 5553152,
        "duration": "361.533333",
        "bit_rate": "13916",
        "bits_per_raw_sample": "8",
        "disposition": {
          "default": 1,
          "dub": 0,
          "original": 0,
          "comment": 0,
          "lyrics": 0,
          "karaoke": 0,
          "forced": 0,
          "hearing_impaired": 0,
          "visual_impaired": 0,
          "clean_effects": 0,
          "attached_pic": 0,
          "timed_thumbnails": 0
        },
        "tags": {
          "creation_time": "2019-09-11T14:36:13.000000Z",
          "language": "und",
          "handler_name": "ISO Media file produced by Google Inc."
        }
      }
    ],
    "copyStreamIdsByFile": {
      "C:\\Test1 vid.mp4": {
        "0": true
      },
      "C:\\Test1 audio.mp4": {},
      "C:\\Test1 audio.mp3": {},
      "C:\\Test1 audio.opus": {
        "0": true
      }
    },
    "cutSegments": [
      {}
    ],
    "fileFormatData": {
      "filename": "C:\\Test1 vid.mp4",
      "nb_streams": 1,
      "nb_programs": 0,
      "format_name": "mov,mp4,m4a,3gp,3g2,mj2",
      "format_long_name": "QuickTime / MOV",
      "start_time": "0.000000",
      "duration": "361.533333",
      "size": "16504121",
      "bit_rate": "365202",
      "probe_score": 100,
      "tags": {
        "major_brand": "dash",
        "minor_version": "0",
        "compatible_brands": "iso6avc1mp41",
        "creation_time": "2019-09-11T14:36:13.000000Z"
      }
    },
    "rotation": 360,
    "shortestFlag": false
  },
  "platform": "win32",
  "version": "3.33.1"
}

**Comment by mifi** [comment]
I believe opus in mp4 is a bit experimental, maybe not all players can handle that

**Reply by NeighbouringPanda** [reply]
ah okay. thx
